### ISSUE TITLE
Zwave Remove Node

### ISSUE THREAD

**Issue by filoor** [athombv/homey#146]
I cant remove a node i try to remove my fibaro motion sensor but it wont remove it ("Picture 1, 2") and then it stays on ("running in another sesion. Try agan later")("Picture 3") must reboot homey before it acts normal and also 
tried to remove it with remove failed node but it keeps loading 4ever ("Picture 4") and also i can add one sensor as match as i want it don't says that it already exist ("Node 7")("Picture 5")

![Picture 1](https://cloud.githubusercontent.com/assets/3647638/13003774/cfce2980-d177-11e5-9d71-713d23706d76.jpg)

![Picture 2](https://cloud.githubusercontent.com/assets/3647638/13003775/cfcfd64a-d177-11e5-8e5c-c609db0211d3.jpg)

![Picture  3](https://cloud.githubusercontent.com/assets/3647638/13004140/edfc6ff0-d179-11e5-9cb8-93ec9fe5a933.jpg)

![Picture 4](https://cloud.githubusercontent.com/assets/3647638/13003886/4dab4518-d178-11e5-8e49-90f4c2d1b200.jpg)

![Picture 5](https://cloud.githubusercontent.com/assets/3647638/13003773/cfcdb130-d177-11e5-934b-46fea66f770a.jpg)

**Comment by JohanKuster** [comment]
Same here with the Fibaro multisensor. Had to reboot. Sensor was not removed.

**Comment by ocatie** [comment]
You can only have one Remove session open. If you refresh the page or open another window you get the message "Running in another session".

I'll look into why the first session won't finish. To reset the process use the Restart Z-Wave Chip and reload the page. If that doesn't help or the button keeps disabled the driver has crashed and you'll have to reboot.

**Comment by RemonBerends** [comment]
Here also impossible to remove a Zwave node.

**Comment by ocatie** [comment]
I've fixed a few driver crashes in 0.8.18 and 0.8.19. Please report back at both versions.

**Comment by RemonBerends** [comment]
When will these updates be released?

**Comment by JohanKuster** [comment]
good question....no answer

**Comment by jghaanstra** [comment]
Reporting back after the upgrade to 0.8.20. The "Remove Failed Node" still isnt working, upon trying nothing happens and the zwave settings page because unresponsive. Trying to reload it only shows the loader icon. The only way to solve this is to pull the plug from Homey.

Using the "Request Network Update" also generates an error.

**Comment by ocatie** [comment]
Could you please powercycle Homey, reproduce the problem and send us the log.

You can send us the log using the button in Settings -> System -> Send Log. A code will be generated that you can place here.

**Comment by jghaanstra** [comment]
@ocatie the log ID is: F4E34D5B3A

What I did before producing this log:
1. power cycle by pulling the plug
2. navigate to zwave settings
3. select a failed node and click the remove failed node button and let this run for about 3 minutes (nothing happens really, only a progress indicator inside the button)
4. refresh the zwave settings page by pressing ctrl + f5 and watching the progress indicator for 2 minutes or so ( the settings page does not load anymore)
5. navigate to the system settings and generate the log

Hope it helps, the device I'm trying to remove is a Greenwave Powernode 6 which has a main device and 6 sockets. The remove failed device option is only available on the main device.

**Comment by ocatie** [comment]
I see that there are 2 powernodes that can't be reached upon startup. This are you ghost nodes if I'm not mistaken. After that I don't see any processes getting triggered in Homey. This explains why nothing happens on the settings page. The button click doesn't trigger the process.

Could you please post a screenshot of your browser console log after pressing the Remove Failed Node button on the settings page?

**Comment by jghaanstra** [comment]
You are indeed correct, both these powernodes have been added again and can be reached under another association but I need to get rid of these failed nodes.

I cleared the console before pressing the remove failed node button as there where some GUI related javascript errors but after pressing the remove failed button nothing was initially registered in the console. Looking at the network tab of the chrome developers console it was stuck pending on this action: http://<<ip>>/api/wireless/zwave/removeFailedNode/2?_=1456508646204. But when coming back from the network tab an error appeared on the screen which I had only seen on 0.18 and not on 0.20 yet and there was some console logging available. After this error I was still able to load the zwave settings as well.

![zwave](https://cloud.githubusercontent.com/assets/4710062/13360506/dec0e62e-dcb9-11e5-8948-549ce0c10671.png)

**Comment by ocatie** [comment]
In 0.8.21 I fixed a bug where the settings wouldn't load when there is a pairing session active.

Could you on 0.8.21 try to reproduce the error after a reboot and send the log.

**Comment by jghaanstra** [comment]
Ok, just tested this on 0.8.22. Here are my findings.

- for the remove failed node button to show I need to refresh the node first to have it flagged as failed. Upon doing this I get an error "Unhandled error Unknown Device" and a 500 error in the console:
`POST http://192.168.0.120/api/wireless/zwave/refresh/2?_=1457419905401 500 (Internal Server Error)window.Homey.api @ api.js:1o.contentWindow.Homey.api @ settings.js:1(anonymous function) @ (index):1Z.event.dispatch @ jquery.js:2m.handle @ jquery.js:2`
- refreshing nodes even for active non-ghost z-wave devices this results in the same error, as I still experience z-wave time-outs this is very bad news for me;
- After trying to refresh a node the list of zwave devices initially stayed empty the time, even upon refreshing the page, restarting the zwave chip, doing a reboot and a power cycle
- Maybe I was impatient but after switching a z-wave socket the list of z-wave devices did appeare again but the ghost device I'm trying to remove is still not flagged as failed so I cant remove it. On other times trying to refresh a failed device the list remained filled.

In short, the refresh node functionality is broken so I'm unable to test the remove failed functionality.

I have send a log with ID: AF2C0DF472

**Comment by Fire69** [comment]
I'm having the same problems.

**Comment by ocatie** [comment]
On boot the devices are are checked if their node id exist in the network. If not a delete button should be available in the right top corner of the device in the zone overview. Is this button present?

**Comment by jghaanstra** [comment]
No, it had the visibility hidden CSS markup. On 0.8.21 there was an asteriks before the devices that where failed in the devices list. Those are gone now so the devices are not on the failed node list. I even removed the visibility: hidden markup and pressed the button on a failed device but the error showed up that the device is not on the failed node list.

**Comment by ocatie** [comment]
I mean in the Zone overview, not the settings page. In the log I see there are devices in Homey that are not in the network. Those devices should now have a delete button in the Zone overview. This button removes the device from Homey (they are not present in the network so the Remove Node wizard can't be used).

**Comment by Fire69** [comment]
Ah, you can just delete a device in the Zone overview!  Didn't know that, is it new?
Still have a problem though.  When I tried adding my Dimmer2, I got an 'unknown node' that I can't remove.
Did a reboot, and a refresh. No way to delete this device.  This is the log: 361743FA71

**Comment by ocatie** [comment]
Yes this is new. Unknown device means the node id is in the network but there is no device in Homey with that id. You can remove the node from the network using the Remove Node button.

**Comment by ocatie** [comment]
Yes this is new. Unknown device means the node id is in the network but there is no device in Homey with that id. You can remove the node from the network using the Remove Node button.

In your log I don't see you starting the removeNode process. Only putting your device in learnmode

**Comment by Fire69** [comment]
True, I didn't start the removeNode process, because I thought that I just had to refresh the node and than a buttton 'remove failed node' should have been visible. :-)
I will try to remove it with removeNode this evening.  (And try to add the Dimmer2 again... )

**Comment by ocatie** [comment]
If the node can't be reached the Remove Failed Node button should pop up. If the refresh is successfull the device object is rebuild in the driver.

**Comment by jghaanstra** [comment]
Awesome, I hadnt noticed the new functionality for removing zwave devices through the devices screen. This got rid of a lot of ghost devices for me, thanx Taco. I do still have an issue though.

I had a ghost device for the greenwave powernode 6 where there where actually overlapping node id's. When adding a greenwave powernode 6 there a 7 devices created, one for each socket and some sort of master device (which actually only switched the first socket). The node numbering for these devices is x = master and x.x is the actual socekt. Because of overlapping node id's I was able to remove the master node (in this case node 5) but I'm now stuck with the socket node id's (5.1,5.2,5.3,5.4,5.5,5.6) and I dont see a way to remove them. In the devices screen they seem operational although greyed out when trying to switch the socket. In the z-wave settings they now appear with a asteriks but there is no remove failed node as this button only appears on the master node, which isnt there anymore.

I hope you understand my fuzzy explanation. Is this something you can work with. I can supply logs and even access if needed.

**Comment by ocatie** [comment]
You should also be able to remove the endpoint devices in the Zone overview. Is the delete button in the top right corner not there? According to your log it should be. Otherwise that is a bug.

**Comment by jghaanstra** [comment]
No, the remove button is not there. These devices actually show as active devices in the zones view and it looks like the delete button on the device cards only show on devices that are not in the network. Other active devices that are actually active dont show this delete button as well and this does seem correct.

My issue is that these endpoint devices actually do not exist as the device has been added under another node ID and works properly there. It seems Homey does not recognize these endpoints as device not in the network or failed devices with the result that I have no way of removing them.

**Comment by ocatie** [comment]
Did you refresh the Zone overview? I can't reproduce endnodes not having the delete button.

**Comment by jghaanstra** [comment]
Perhaps a picture says a 1000 words. On image one you see the devices under the zwave settings with the asteriks and marked as failed nodes. Selecting them wont bring up the remove failed node button. As far as I know this button only appears on the main node, which is not there anymore (node 5). On picture two you see the devices of the endNodes in the zone view, they look like active devices and I can even edit their settings but there is no physical device connected to these devices and there is no delete button (as the devices seem to be registered as active devices under the zone view).

![01-settings](https://cloud.githubusercontent.com/assets/4710062/13607372/20a22082-e550-11e5-9750-b2879dbe34e6.png)
![02-devices](https://cloud.githubusercontent.com/assets/4710062/13607378/27793f9e-e550-11e5-97e5-b93c997588f1.png)

**Comment by ocatie** [comment]
Could you please reboot and send a log?

**Comment by jghaanstra** [comment]
Sure thing, here you go: 8DAEE6EA65

If there is anything I can do let me know (like giving access to my Homey).

**Comment by ocatie** [comment]
As of 0.8.27 we've got a rewritten settings page. Please create new issue for version 0.8.27+
